auth_mellon_util.c

```diff
@@ -574,16 +574,25 @@ int am_read_post_data(request_rec *r, char **data, apr_size_t *length)
     bytes_left = len;
 
     while (bytes_left > 0) {
-        /* Read data from the client. Returns 0 on EOF or error, the
-         * number of bytes otherwise.
+        /* Read data from the client. Returns 0 on EOF and -1 on
+         * error, the number of bytes otherwise.
          */
         read_length = ap_get_client_block(r, &(*data)[bytes_read],
                                           bytes_left);
         if (read_length == 0) {
+            // got the EOF
+            break;
+        }
+        else if (read_length < 0) {
             ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r,
                           "Failed to read POST data from client.");
             return HTTP_INTERNAL_SERVER_ERROR;
         }
+        else if ((apr_size_t)read_length > bytes_left) {
```

Comment:
Could you clarify this a bit -- in what cases will `ap_get_client_block` return more than `bytes_left`?

From the [documentation](https://github.com/apache/httpd/blob/2.4.18/include/http_protocol.h#L494), it looks like it shouldn't do that:
```
/**
 * Call this in a loop.  It will put data into a buffer and return the length
 * of the input block
 * @param r The current request
 * @param buffer The buffer in which to store the data
 * @param bufsiz The size of the buffer
 * @return Number of bytes inserted into the buffer.  When done reading, 0
 *         if EOF, or -1 if there was an error
 */
```